jenkins_node_scanner.py

```diff
@@ -220,18 +236,23 @@ def main():
                 UNPARSEABLE_NODES_FOUND.labels(args.url).set(
                     len(raw_node_ips) - len(node_ips))
 
-                node_info = [{
-                    'labels': {
-                        'jenkins_master': args.url,
-                        'node': node['name'],
-                    },
-                    'targets': ['%s:%d' % (ip, args.target_port)],
-                } for node, ip in node_ips]
+                node_info = []
+
+                for node, ip_addr in node_ips:
+                    node_info.append({
+                        'labels': {'jenkins_master': args.url, 'node': node['name']},
+                        'targets': ['%s:%d' % (ip_addr, port) for port in args.target_port
+                                    if test_port(ip_addr, port)],
```

Comment:
If neither port is available, it will just emit an empty `targets` field. I guess that's okay.